packages/terra-paginator/src/PageLabel.js

```diff
@@ -0,0 +1,22 @@
+export default function getPageLabel(pageLabel, selectedPage, totalPages) {
+  let messageId;
+  let messageAttributes;
+
+  if (totalPages) {
+    if (pageLabel) {
+      messageId = 'Terra.paginator.pageCountWithLabel';
+      messageAttributes = { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages };
+    } else {
+      messageId = 'Terra.paginator.pageCount';
+      messageAttributes = { pageNumber: selectedPage, pageNumberTotal: totalPages };
+    }
+  } else if (pageLabel) {
+    messageId = 'Terra.paginator.pageIndexWithLabel';
+    messageAttributes = { pageLabel, pageNumber: selectedPage };
+  } else {
+    messageId = 'Terra.paginator.pageIndex';
+    messageAttributes = { pageNumber: selectedPage };
+  }
+
+  return { messageId, messageAttributes };
+}
```

Comment:
Totally stylistic but you could write it like this too.
```suggestion
export default function getPageLabel(pageLabel, selectedPage, totalPages) {
  if (totalPages && pageLabel) {
    return ({
      messageId: 'Terra.paginator.pageCountWithLabel',
      messageAttributes: { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages },
    })
   if (totalPages) {
      return ({
        messageId: 'Terra.paginator.pageCount',
        messageAttributes: { pageNumber: selectedPage, pageNumberTotal: totalPages },
      )}
    }
  if (pageLabel) {
    return ({
      messageId: 'Terra.paginator.pageIndexWithLabel',
      messageAttributes: { pageLabel, pageNumber: selectedPage },
    })
  }
  return ({
    messageId: 'Terra.paginator.pageIndex',
    messageAttributes: { pageNumber: selectedPage },
  });
}
```